src/payments/services/payments.service.ts

```
@@ -53,4 +56,87 @@ export class PaymentsService {
 
     return { paymentIntent, user }
   }
+
+  async updateMissingCustomerId(email: string) {
+    const user = await this.usersService.findUserByEmail(email)
+    if (!user) {
+      return null
+    }
+
+    if (user.metaData?.customerId) {
+      return null
+    }
+
+    const checkoutSessionId = user.metaData?.checkoutSessionId as string
+    if (!checkoutSessionId) {
+      return null
+    }
+    const customerId =
+      await this.stripe.fetchCustomerIdFromCheckoutSession(checkoutSessionId)
+    if (!customerId) {
+      return null
+    }
+
+    await this.usersService.patchUserMetaData(user!.id, {
+      customerId,
+      checkoutSessionId: null,
+    })
+    return user
+  }
+
+  async fixMissingCustomerIds() {
+    const users = await this.usersService.findMany({
+      where: {
+        metaData: {
+          path: ['customerId'],
+          equals: Prisma.JsonNull,
+        },
+        AND: {
+          metaData: {
+            path: ['checkoutSessionId'],
+            not: Prisma.JsonNull,
+          },
+        },
+      },
+      select: {
+        email: true,
+        id: true,
+      },
+    })
+
+    const results: {
+      success: string[]
+      failed: { email: string; error: string }[]
+      skipped: string[]
+    } = {
+      success: [],
+      failed: [],
+      skipped: [],
+    }
+
+    for (const dbUser of users) {
+      const { email } = dbUser
+      try {
+        const user = await this.updateMissingCustomerId(email)
+        if (!user) {
+          results.skipped.push(email)
+          continue
```

idk if it's just early in AM, but I think failed will never make it's way into the results array, only successes. You would need to throw an error and continue that way.